Sources/Layout/FlexLayout.swift

```diff
@@ -50,46 +50,39 @@ public class FlexLayout<Data>: AxisDependentLayout<Data> {
     self.insets = insets
   }
 
-
   public override func layout(collectionSize: CGSize,
                               dataProvider: CollectionDataProvider<Data>,
-                              sizeProvider: CollectionSizeProvider<Data>) -> [CGRect] {
+                              sizeProvider: @escaping CollectionSizeProvider<Data>) -> [CGRect] {
     var frames: [CGRect] = []
-    var totalHeight: CGFloat = padding * CGFloat(dataProvider.numberOfItems - 1)
-    var totalFlex: CGFloat = 0
-    for i in 0..<dataProvider.numberOfItems {
-      if let flex = flex[dataProvider.identifier(at: i)] {
-        totalHeight += flex.range.lowerBound
-        totalFlex += flex.flex
-      } else {
-        let size = sizeProvider(i, dataProvider.data(at: i), collectionSize)
-        totalHeight += primary(size)
-      }
-    }
 
-    let heightPerFlex = totalFlex > 0 ? (primary(collectionSize) - totalHeight) / totalFlex : 0
+    let (sizes, totalPrimary) = getSizes(collectionSize: collectionSize,
+                                         dataProvider: dataProvider,
+                                         sizeProvider: sizeProvider)
 
     var offset: CGFloat = 0
-    if totalHeight < primary(collectionSize), totalFlex == 0 {
+    var padding = self.padding
+    if totalPrimary < primary(collectionSize) {
+      let leftOverPrimaryWithPadding = primary(collectionSize) - totalPrimary
+      let leftOverPrimary = leftOverPrimaryWithPadding + self.padding * CGFloat(dataProvider.numberOfItems - 1)
```

Comment:
leftOverPrimary = leftOverPrimaryWithPadding - self.padding * CGFloat(dataProvider.numberOfItems - 1)?

Reply:
TotalPrimary is calculated from all cell height + all minimum paddings inbetween. Therefore if we want to remove the paddings from our calculation we need to subtract it by (count - 1) * minimum padding

Reply:
gotcha